src/fs/read_dir.rs

```diff
@@ -93,28 +143,31 @@ pub async fn read_dir<'js>(
             }
         }
     };
-    let mut items = Vec::with_capacity(64);
 
-    let mut directory_walker = DirectoryWalker::new(PathBuf::from(path), |_| true);
+    let mut directory_walker: DirectoryWalker<fn(&str) -> bool> =
```

Comment:
Do we need explicit type here?

Reply:
I tried to type the closure instead but the compiler won't let me (I had to put the explicit type in there for it to compile)